packages/pelagos/src/treeView/TreeNode.js

```diff
@@ -145,15 +151,19 @@ const TreeNode = ({id, labelClassName, label, icon: Icon, expanded, loading, chi
 		[expanded, level, onSelect, onToggle, path, selected, setFocused]
 	);
 
+	if (!expanded && focused.length > childrenLevel && arrayEquals(focused.slice(0, childrenLevel), path)) {
+		setFocused(path);
+	}
+
 	return (
 		<li
 			{...props}
 			id={id}
 			className="TreeView__node"
-			tabIndex={focused === id ? 0 : -1}
+			tabIndex={arrayEquals(focused, path) ? 0 : -1}
```

Comment:
This made me think about the need of arrayEquals...
If every node has to have a unique ID, why do we need to compare the full arrays?

Would this not work:
```
tabIndex={focused.at(-1) === path.at(-1) ? 0 : -1}
```

If that is not wrong, I think all usages could also be replaced? For example in:
```
if (!expanded && focused.length > childrenLevel && arrayEquals(focused.slice(0, childrenLevel), path)) {
```
Could it be:
```
if (!expanded && focused.length > childrenLevel && focused.at(level) === path.at(-1)) {
```

Reply:
I guess `path.at(-1)` is always `id`, so that could even be simplified more (assuming I am not way off)